internal/setup/setup.go

```diff
@@ -583,7 +614,19 @@ func parsePackage(baseDir, pkgName, pkgPath string, data []byte) (*Package, erro
 			var mutable bool
 			var until PathUntil
 			var arch []string
-			if strings.ContainsAny(contPath, "*?") {
+			var generate GenerateKind
+			if yamlPath != nil && yamlPath.Generate != "" {
+				zeroPathGenerate := zeroPath
+				zeroPathGenerate.Generate = yamlPath.Generate
+				if !yamlPath.SameContent(&zeroPathGenerate) || yamlPath.Until != UntilNone {
+					return nil, fmt.Errorf("slice %s_%s path %s has invalid generate options",
+						pkgName, sliceName, contPath)
+				}
+				if _, err := GetGeneratePath(contPath); err != nil {
+					return nil, fmt.Errorf("slice %s_%s has %s", pkgName, sliceName, err)
```

Comment:
This error message will be bogus depending on the error returned by the function.  That's why we typically don't embed an unknown error as part of a sentence.

Reply:
Changed in https://github.com/canonical/chisel/pull/143/commits/426ca94d0431f70699d92b2700bff537014d8cf5 to create a self-contained error message.